data.py

```diff
@@ -212,21 +214,27 @@ class RefuelingTextFileConversion(Conversion):
     refuelings: RefuelingRepository
 
     def convert(self) -> None:
-        refuelings = self.refuelings_text_file.read()
-        for line in refuelings:
+        text = self.refuelings_text_file.read()
+        for line in text:
+            line = line[:-1]
             data = line.split('\t')
-            id = int(data[0])
-            date = datetime(data[1])
-            volume = float(data[2])
-            destination_id = int(data[3])
-            new_refueling = Refueling(id, date, volume, destination_id)
-            self.refuelings.add(new_refueling)
+            date = datetime.strptime(data[0], "%Y/%m/%d")
+            volume = float(data[1])
+            destination = next(destination for destination in destinations if destination.name == data[2])
```

Comment:
Repositories shouldn't depend on each other. If you have such dependencies on repository layer, it means that something's wrong with your model. First of all, you should store there ID of destination entry - that's enough to retrieve the destination later, on service level. Secondly, the `next` call is super-suspicious.  Don't depend on order of entries in the file - mark them with IDs and possibly create another file which will store the ID of successor for each destination ID. Otherwise, you can get lost easily when switching to more sophisticated data storages like SQL.

Reply:
OK, I will change destination.name to destination.id in all .txt files and all conversion classes. About next(): In this case syntax in (..) creates a generator object finding an instance of Destination class that has the same name as data in .txt file. The next() function calls the iterator of that object and finds first instance. I thought that it could be nice because of saving memory and computing time. But I think that when switching to ID instead of names it will be redundant.

Reply:
I misunderstood the `next(...)` function. I thought it gets the next element in the iterator after meeting the condition. Why didn't they just call it `first(...)` ...

Reply:
Just store the ID and then on service-level (layer above repository) you can do sth like this:
```python
class RefuelingService:
  def __init__(refueling_repository, destinations_repository):
    self.refueling_repository = refueling_repository
    self.destinations_repository = destinations_repository
  def get_destination(refueling):
    return self.destinations_repository.get_by_id(refueling.destination_id) # get_by_id is useful method to have in repository
```